### DISCUSSION TITLE
Do numpad keys register both the number and the function with NumLock on?

### DISCUSSION THREAD

**Discussion by LB767** [rust-windowing/winit#4030] [Q&A] [answered]
If I have NumLock on and press "1", does "End" also get registered? This is the behavior I'm seeing and it is confusing to me so I wanted to make sure it's intended.

Thanks

**Comment by madsmtm** [comment, marked as answer]
What happens in the example in https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code? 'cause that's roughly the intended behaviour.

**Reply by LB767** [reply]
Ok yeah, after further debugging I now understand this concept of physical and logical keys better. Seems the app level library I'm using is just handling winit wrong and is basically turning a keypress into 2 events, as if there were two distinct physical and logical events...
Problem solved, thanks for the pointers :)
